transports/tcp/src/lib.rs

```diff
@@ -716,6 +636,20 @@ where
                 .unregister(self.listen_addr.ip(), self.listen_addr.port()),
         }
     }
+
+    /// Close the listener.
+    ///
+    /// This will create a [`TransportEvent::ListenerClosed`] event
+    /// and terminate the stream once all remaining events in queue have
+    /// been reported.
+    fn close(&mut self, reason: Result<(), io::Error>) {
+        self.queued_events
+            .push_back(TransportEvent::ListenerClosed {
+                listener_id: self.listener_id,
+                reason,
+            });
```

Comment:
Do we ever push something else in here? If not, then a more specific solution would be:

```rust
enum State {
   Active,
   Closing,
   Closed
}
```

In `Closing` we can then report the `Closed` event, set ourselves to `Closed` and return `None`.

Reply:
It is now an `Option`. It is aligned with at least `quic` and `webrtc`; I did not see such an approach with enums in other transports...